test/e2e/common_test.go

```diff
@@ -136,53 +147,29 @@ func (p *PodmanTestIntegration) Setup() {
 	p.ArtifactPath = ARTIFACT_DIR
 }
 
-// var _ = BeforeSuite(func() {
-// 	cwd, _ := os.Getwd()
-// 	INTEGRATION_ROOT = filepath.Join(cwd, "../../")
-// 	podman := PodmanTestCreate("/tmp")
-// 	podman.ArtifactPath = ARTIFACT_DIR
-// 	if _, err := os.Stat(ARTIFACT_DIR); os.IsNotExist(err) {
-// 		if err = os.Mkdir(ARTIFACT_DIR, 0777); err != nil {
-// 			fmt.Printf("%q\n", err)
-// 			os.Exit(1)
-// 		}
-// 	}
-// })
-// for _, image := range CACHE_IMAGES {
-// 	if err := podman.CreateArtifact(image); err != nil {
-// 		fmt.Printf("%q\n", err)
-// 		os.Exit(1)
-// 	}
-// }
-// host := GetHostDistributionInfo()
-// if host.Distribution == "rhel" && strings.HasPrefix(host.Version, "7") {
-// 	f, err := os.OpenFile("/proc/sys/user/max_user_namespaces", os.O_WRONLY, 0644)
-// 	if err != nil {
-// 		fmt.Println("Unable to enable userspace on RHEL 7")
-// 		os.Exit(1)
-// 	}
-// 	_, err = f.WriteString("15000")
-// 	if err != nil {
-// 		fmt.Println("Unable to enable userspace on RHEL 7")
-// 		os.Exit(1)
-// 	}
-// 	f.Close()
-// }
-// path, err := ioutil.TempDir("", "libpodlock")
-// if err != nil {
-// 	fmt.Println(err)
-// 	os.Exit(1)
-// }
-// LockTmpDir = path
-// })
-
-var _ = AfterSuite(func() {
-	sort.Sort(testResultsSortedLength{testResults})
-	fmt.Println("integration timing results")
-	for _, result := range testResults {
-		fmt.Printf("%s\t\t%f\n", result.name, result.length)
-	}
-})
+var _ = SynchronizedAfterSuite(func() {},
+	func() {
+		sort.Sort(testResultsSortedLength{testResults})
+		fmt.Println("integration timing results")
+		for _, result := range testResults {
+			fmt.Printf("%s\t\t%f\n", result.name, result.length)
+		}
+
+		// previous crio-run
+		tempdir, err := CreateTempDirInTempDir()
+		if err != nil {
+			os.Exit(1)
+		}
+		podmanTest := PodmanTestCreate(tempdir)
+
+		if err := os.RemoveAll(podmanTest.CrioRoot); err != nil {
+			fmt.Printf("%q\n", err)
+		}
+
+		// for localized tests, this removes the image cache dir and for remote tests
+		// this is a no-op
+		removeCache()
```

Comment:
I'm assuming the reason this is a no-op is the remote client can't/shouldn't fuss with storage on that level.

For the "remote" tests, if a script in CI were to set it up, then could the remote tests take advantage of the caching scheme (assuming they're told where to look)?

Reply:
not yet